### PR TITLE
windows build failed in speedb_paired_bloom_internal files due to bad…

### PR DESCRIPTION
**Pull request by ayulas**
… used of double / uint32 casting #383

### CODE REVIEW

**Review by udi-speedb — DISMISSED** [review]

File: port/win/port_win.h [hunk 1]
```diff
@@ -168,6 +168,8 @@ class RWMutex {
   SRWLOCK srwLock_;
 };
 
+#define RWMutexWr RWMutex
```
Comment (udi-speedb) [inline port/win/port_win.h:171]: Why don't you use:
`using RWMutexWr = RWMutex;`

And please also add a comment to explain why that is needed / what is RWMutexWr.
And, is RWMutexWr not needed in Windows as it was needed in Posix-based environments?

**Comment by udi-speedb** [comment]
@ayulas - A preliminary question: Is this PR ready for review with all of the planned changes?
Does it build successfully as it is on Windows?
Do we run make check (or equivalent) on Windows and all of the unit tests pass?
Same question for additional QA tests on Windows?

File: plugin/speedb/memtable/hash_spd_rep.cc [hunk 2]
```diff
@@ -382,22 +383,19 @@ void SpdbVectorContainer::SeekIter(const IterAnchors& iter_anchor,
 }
 
 void SpdbVectorContainer::Merge(std::list<SpdbVectorPtr>::iterator& begin,
-                                std::list<SpdbVectorPtr>::iterator& end) {
+                                std::list<SpdbVectorPtr>::iterator& end,
+                                size_t total_merge_vector_elements) {
```
Comment (udi-speedb) [inline plugin/speedb/memtable/hash_spd_rep.cc:387]: Is adding a parameter to the method part of passing the build on Windows or is it related to some other issue?

**Comment by udi-speedb** [comment]
Are all of the changes needed to successfully build on Windows or does this PR contain functional changes that are not related to being able to successfully compile / build on Windows?

**Review by udi-speedb — CHANGES_REQUESTED** [review]

**Comment by ayulas** [comment]
i wouldnt push and send to qa and run CI if it wasnt built successfully and got approve from max
i took this ticket to help max compilation issue. its not related to qa test and it was supposed handle this.
all the changes here are related to compile errors and warnings

File: plugin/speedb/memtable/hash_spd_rep.cc (hunk 2 above)
Comment (ayulas) [reply plugin/speedb/memtable/hash_spd_rep.cc:387]: compile issue

Comment (ayulas) [reply plugin/speedb/memtable/hash_spd_rep.cc:387]: @hilikspdb pls review the memtable windows compile changes

**Review by Yuval-Ariel — APPROVED** [review]

**Review by Yuval-Ariel — APPROVED** [review]

**Pull request merged**